java/src/main/java/org/digma/intellij/plugin/idea/psi/java/JavaEndpointNavigationProvider.java

```diff
@@ -140,7 +171,7 @@ public void fileChanged(VirtualFile virtualFile) {
                     //if file moved then removeDocumentEndpoints will not remove anything but building endpoint locations will
                     // override the entries anyway
                     removeDocumentEndpoint(virtualFile);
-                    buildEndpointAnnotations(GlobalSearchScope.fileScope(project, virtualFile));
+                    buildEndpointAnnotations(virtualFile);
                 } finally {
                     buildLock.unlock();
```

Comment:
need to check if buildLock.isHeldByCurrentThread

Reply:
and if not?
can you suggest the code block?

Reply:
if (buildLock.isHeldByCurrentThread) {
        lock.unlock()
}

we had exceptions in the past because of that, its just good practice. and if there is code we don't check that then we should.

Reply:
1) its out of scope of this PR
2) do you have links to issue(s)?

Reply:
there are no issues but we had exceptions and in most places we do this check before unlock